embedchain/vectordb/lancedb.py

```diff
@@ -0,0 +1,281 @@
+            self.collection = self.client.create_table(table_name, schema=schema)
+
+        self.collection = self.client[table_name]
+
+        return self.collection
+
+    def get(self, ids: Optional[List[str]] = None, where: Optional[Dict[str, any]] = None, limit: Optional[int] = None):
+        """
+        Get existing doc ids present in vector database
+
+        :param ids: list of doc ids to check for existence
+        :type ids: List[str]
+        :param where: Optional. to filter data
+        :type where: Dict[str, Any]
+        :param limit: Optional. maximum number of documents
+        :type limit: Optional[int]
+        :return: Existing documents.
+        :rtype: List[str]
+        """
+        if limit is not None:
+            max_limit = limit
+        else:
+            max_limit = 3
+        results = {"ids": [], "metadatas": []}
+
+        where_clause = {}
+        if where:
+            where_clause = self._generate_where_clause(where)
+
+        if ids is not None:
+            records = (
+                self.collection.to_lance().scanner(filter=f"id IN {tuple(ids)}", columns=["id"]).to_table().to_pydict()
+            )
+            for id in records["id"]:
+                if where is not None:
+                    result = (
+                        self.collection.search(query=id, vector_column_name="id")
+                        .where(where_clause)
+                        .limit(max_limit)
+                        .to_list()
+                    )
+                else:
+                    result = self.collection.search(query=id, vector_column_name="id").limit(max_limit).to_list()
+                results["ids"] = [r["id"] for r in result]
+                results["metadatas"] = [r["metadata"] for r in result]
+
+        return results
+
+    def add(
+        self,
+        documents: List[str],
+        metadatas: List[object],
+        ids: List[str],
+    ) -> Any:
+        """
+        Add vectors to lancedb database
+
+        :param documents: Documents
+        :type documents: List[str]
+        :param metadatas: Metadatas
+        :type metadatas: List[object]
+        :param ids: ids
+        :type ids: List[str]
+        """
+        data = []
+        to_ingest = list(zip(documents, metadatas, ids))
+        for doc, meta, id in to_ingest:
+            temp = {}
+            temp["doc"] = doc
+            temp["vector"] = self.embedder.embedding_fn([doc])[0]
+            temp["metadata"] = str(meta)
+            temp["id"] = id
+            data.append(temp)
+
+        self.collection.add(data=data)
+
+    def _format_result(self, results) -> list:
+        """
+        Format LanceDB results
+
+        :param results: LanceDB query results to format.
+        :type results: QueryResult
+        :return: Formatted results
+        :rtype: list[tuple[Document, float]]
+        """
+        return results.tolist()
+
+    def query(
+        self,
+        input_query: list[str],
```

Comment:
Please make it a `str`. It has been changed for all vectordbs here #1388

Reply:
ok